guides/keras_hub/object_detection_retinanet.py

```diff
@@ -0,0 +1,750 @@
+        result[key] = values
+
+    # The ragged objects need some special handling
+    for key in ["label", "bbox"]:
+        values = []
+        objects = [value["objects"] for value in metadata]
+        for object in objects:
+            values.append([o[key] for o in object])
+        result["objects/" + key] = values
+    return result
+
+
+def build_dataset_from_metadata(metadata):
+    """Builds TensorFlow dataset from the image metadata of VOC dataset."""
+    # The objects need some manual conversion to ragged tensor.
+    metadata["labels"] = tf.ragged.constant(metadata["labels"])
+    metadata["objects/label"] = tf.ragged.constant(metadata["objects/label"])
+    metadata["objects/bbox"] = tf.ragged.constant(
+        metadata["objects/bbox"], ragged_rank=1
+    )
+
+    dataset = tf.data.Dataset.from_tensor_slices(metadata)
+    dataset = dataset.map(load_images, num_parallel_calls=tf.data.AUTOTUNE)
+    return dataset
+
+
+def load_voc(
+    year="2007",
+    split="trainval",
+    data_dir="./",
+    voc_url=VOC_2007_URL,
+):
+    extracted_dir = os.path.join("VOCdevkit", f"VOC{year}")
+    get_data = keras.utils.get_file(
+        fname=os.path.basename(voc_url),
+        origin=voc_url,
+        cache_dir=data_dir,
+        extract=True,
+    )
+    data_dir = os.path.join(get_data, extracted_dir)
+    image_ids = get_image_ids(data_dir, split)
+    metadata = build_metadata(data_dir, image_ids)
+    dataset = build_dataset_from_metadata(metadata)
+
+    return dataset
+
+
+"""
+## Load the dataset
+"""
+train_ds_2007 = load_voc(
+    year="2007",
+    split="trainval",
+    data_dir="./",
+    voc_url=VOC_2007_URL,
+)
+train_ds_2012 = load_voc(
+    year="2012",
+    split="trainval",
+    data_dir="./",
+    voc_url=VOC_2012_URL,
+)
+
+"""
+## Inference using a pre-trained object detector
+
+Let's begin with the simplest `KerasHub` API: a pre-trained object detector. In
+this example, we will construct an object detector that was pre-trained on the
+`COCO` dataset. We'll use this model to detect objects in a sample image.
+
+The highest-level module in KerasHub is a `task`. A `task` is a `keras.Model`
+consisting of a (generally pre-trained) backbone model and task-specific layers.
+Here's an example using `keras_hub.models.ImageObjectDetector` with the
+`RetinaNet` model architecture and `ResNet50` as the backbone.
+
+ResNet is a great starting model when constructing an image classification
+pipeline. This architecture manages to achieve high accuracy while using a
+relatively small number of parameters. If a ResNet isn't powerful enough for the
+task you are hoping to solve, be sure to check out KerasHub's other available
+backbones here https://keras.io/keras_hub/presets/
+"""
+
+object_detector = keras_hub.models.ImageObjectDetector.from_preset(
+    "kaggle://keras/retinanet/keras/retinanet_resnet50_fpn_coco"
+)
+object_detector.summary()
+
+"""
+## Preprocessing Layers
```

Comment:
add a single line or a few lines of description of what this section will be about

Reply:
Done